### DISCUSSION TITLE
Update image if tag updated in helm charts

### DISCUSSION THREAD

**Discussion by ayushkedia357** [fluxcd/flux2#2242] [Q&A] [answered]
Want to automate git updates on k8s cluster using fluxcd. For each revision in the helm charts sourced from git rep flux will reconcile. Updates related to the environment values, replica changes in the charts gets updated in the cluster as well. But this is not working when I update the image tag in the values file. The image is to be pulled from a private docker-registry. 

Need help with the above.

`flux: v0.24.0`
`kubernetes: v1.22.3`

**Comment by kingdonb** [comment, marked as answer]
When you have a chart hosted in a gitrepository, the `Chart.yaml` which mentions a `version` needs to change or there is no new revision in the default configuration of a HelmRelease.

If you set `spec.chart.spec.reconcileStrategy` to `Revision` instead of the default `ChartVersion` it should behave as you're expecting. The problem is that HelmChart versions are meant to be immutable, so when you add a new commit to your repo, but don't increment the version number, HelmRelease and source controller will not produce any updates. They need to see a new `HelmChart` revision in order to take action. By default the version / revision is derived from the Chart Version, but you can switch it to using git SHA as a revision instead.

In general the experience of working with Helm charts in GitRepository leaves a lot to be desired, as I expect you'll find for yourself... I strongly recommend using HelmRepository in production instead for now at least. HelmRepo has all of the expected semantics for distributing Helm charts according to Helm, where there are definitely some idiosyncrasies when using Git to serve up Helm charts. GitRepo charts are a nice feature for development, but I wouldn't consider them as a replacement for versioning and tagging with proper release cycling.
